docs/source/platform/schema-validation.mdx

```diff
@@ -12,18 +16,24 @@ Apollo provides a tool to protect for exactly this scenario called **schema vali
 ## How it works
 
 Schema validation is run through the Apollo CLI by executing the `apollo service:check` command. Apollo will generate a diff between your local schema and your most recently registered schema, then validate that the changes are safe by checking if any queries actively running against your graph will be affected.
+If your graph is composed using federation, you can also use the `apollo service:check` command to validate individual service changes by adding the `--serviceName` flag. However, unlike running service check on a non-federated service, Engine first needs to run composition on the proposed partial schema with the current list of implementing services and their partial schemas. Composition must pass before validation can continue.
+
+But wait... What exactly is composition? 🤔 In order to have different implementing services contribute to the same graph, there needs to be a way to combine each implementing service's partial schemas into one large data graph. Engine's composition algorithm does exactly this.
 
-Here's how it works:
+Here's how schema validation works for federated and non-federated services:
 
-1. You run `apollo service:check` locally or in CI. The proposed schema is sent to Engine's schema registry.
-1. Engine creates a diff between the local schema and the most recently published schema in the registry.
+1. You run `apollo service:check` locally or in CI. For federated services run the same command but add `--serviceName=ExampleService --endpoint=http://loocalhost:4000/graphql` to the end. The proposed schema/partial schema (if using federation) is sent to Engine's schema registry.
+1. If you're using federation, Engine's schema registry will attempt to compose the partial schema with the current schema. If composition fails then we do not continue with validation and exit early.
+1. Engine creates a diff between the local schema and the most recently published schema in the registry. For federation this step happens after composition passes.
 1. Engine fetches a list of all operations sent to your graph in the last day (time window is [configurable](#adjusting-validation-parameters)).
 1. Engine walks through the schema diff change-by-change and compares against the operation list to see if the changes will affect the behavior of any operations.
 1. Engine returns the schema diff and indicates any breaking changes found.
 1. The CLI prints the output of this check with a link to view more details in the Engine UI.
 
 ### Breaking change detection
 
+Since federation involves merging the partial schema you're running service check against with the current schema, it's possible to run into compposition errors. For example, if you try to run validation on a partial schema with conflicting type names as the current schema then this will definitely result in a composition error. 
```

Comment:
```suggestion
The first breaking change Engine can capture is a federated service being unable to compose with the rest of a federated graph. Types of composition failures are documented [here]().
```

I feel like there's a lot of "because of this" that we don't need. I think this calls for more of the _what_ than the _why_.

- [ ] This needs the link to the composition failure docs

Reply:
where are the composition failures doc?